### PR TITLE
Parser test

### LINKED ISSUE #176: AddContactCommand

### LINKED ISSUE #177: DeleteContactCommand

### LINKED ISSUE #179: AddContactCommandTest

### LINKED ISSUE #180: DeleteContactCommandTest

### PR DESCRIPTION
**Pull request by KevinLohJunYong**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -109,7 +157,7 @@ public static Tag parseTag(String tag) throws ParseException {
         requireNonNull(tag);
         String trimmedTag = tag.trim();
         if (!Tag.isValidTagName(trimmedTag)) {
-            throw new ParseException(String.format("Tag given: %s\n%s", trimmedTag, Tag.MESSAGE_CONSTRAINTS));
+            throw new ParseException("Tags names should be alphanumeric");
```
Comment (lrj689) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Why was this changed?

File: src/main/java/seedu/address/logic/commands/AddContactCommand.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            + "[" + PREFIX_TAG + "TAG]...\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "Bob Choo "
+            + PREFIX_PHONE + "22222222 "
+            + PREFIX_EMAIL + "bob@example.com "
+            + PREFIX_TAG + "friends "
+            + PREFIX_TAG + "owesMoney";
+
+    public static final String MESSAGE_SUCCESS = "New contact added: %1$s";
+    public static final String MESSAGE_DUPLICATE_CONTACT = "This contact already exists in Teaching Assistant";
```
Comment (goatygoatygoat) [inline src/main/java/seedu/address/logic/commands/AddContactCommand.java]: This should be in `Messages` class in commons/core

File: src/main/java/seedu/address/logic/commands/exceptions/CommandException.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package seedu.address.logic.commands.exceptions;
 
 /**
- * Represents an error which occurs during execution of a {@link Command}.
+ * Represents an error which occurs during execution of a command
```
Comment (lrj689) [inline src/main/java/seedu/address/logic/commands/exceptions/CommandException.java]: Why did you change this?

**Review by lrj689 — CHANGES_REQUESTED** [review]
Why are you changing things that are not involved with add contact and delete contact??

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (goatygoatygoat) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java]: Please change back.

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalEntries.getTypicalEntriesList;
```
Comment (goatygoatygoat) [inline src/test/java/seedu/address/logic/commands/AddContactCommandTest.java]: Why is this imported

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.contact.Contact;
+import seedu.address.testutil.ContactBuilder;
+
+public class AddContactCommandTest {
+    private ModelManager model = new ModelManager(getTypicalEntriesList(), new UserPrefs());
```
Comment (goatygoatygoat) [inline src/test/java/seedu/address/logic/commands/AddContactCommandTest.java]: Not sure why you are using getTypicalEntriesList() when you are testing for a contact command.

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    @Test
+    public void constructor_nullcontact_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddContactCommand(null));
+    }
+
+    @Test
+    public void execute_contactAcceptedByModel_addSuccessful() throws Exception {
```
Comment (goatygoatygoat) [inline src/test/java/seedu/address/logic/commands/AddContactCommandTest.java]: No exception should be thrown if contact is accepted by model.

File: src/main/java/seedu/address/logic/commands/AddContactCommand.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.contact.Contact;
+
+/**
+ * Adds a contact to Teaching Assistant.
+ */
+public class AddContactCommand extends Command {
+
+    public static final String COMMAND_WORD = "add";
```
Comment (tsh22) [inline src/main/java/seedu/address/logic/commands/AddContactCommand.java:18]: Command word should be "cadd", have to handle this commandword in the AddressBookParser also

File: src/main/java/seedu/address/logic/commands/DeleteContactCommand.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.contact.Contact;
+
+/**
+ * Deletes a contact identified using it's displayed index from the address book.
+ */
+public class DeleteContactCommand extends Command {
+
+    public static final String COMMAND_WORD = "delete";
```
Comment (tsh22) [inline src/main/java/seedu/address/logic/commands/DeleteContactCommand.java:18]: Command word should be "cdelete", have to handle this commandword in the AddressBookParser also

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 9]
```diff
@@ -51,6 +54,21 @@ public static Name parseName(String name) throws ParseException {
...
+     * Parses a {@code String name} into a {@code Name}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code name} is invalid.
+     */
+    public static ContactName parseContactName(String name) throws ParseException {
+        requireNonNull(name);
+        String trimmedName = name.trim();
+        if (!ContactName.isValidName(trimmedName)) {
+            throw new ParseException(ContactName.MESSAGE_CONSTRAINTS);
```
Comment (tsh22) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Just merged my ParserUtil.java which also implements these functions. Can solve merge conflicts to ensure parser for ContactName, Phone, Email throws ParseException in the form:  throw new ParseException(String.format("<field> given: %s\n%s", trimmed<field>, <field>.MESSAGE_CONSTRAINTS));

File: src/main/java/seedu/address/logic/commands/AddContactCommand.java (hunk 7 above)
Comment (tsh22) [inline src/main/java/seedu/address/logic/commands/AddContactCommand.java]: Command word should be "cadd", have to handle this commandword in the AddressBookParser also

File: src/main/java/seedu/address/logic/commands/DeleteContactCommand.java (hunk 8 above)
Comment (tsh22) [inline src/main/java/seedu/address/logic/commands/DeleteContactCommand.java]: Command word should be "cdelete", have to handle this commandword in the AddressBookParser also

**Review by tsh22 — CHANGES_REQUESTED** [review]

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        Contact validContact = new ContactBuilder().build();
+        CommandResult commandResult = new AddContactCommand(validContact).execute(model);
+        String feedBack = commandResult.getFeedbackToUser();
+        assertEquals(String.format(AddContactCommand.MESSAGE_SUCCESS, validContact), feedBack);
+    }
+    @Test
+    public void execute_duplicateContact_throwsCommandException() {
```
Comment (goatygoatygoat) [inline src/test/java/seedu/address/logic/commands/AddContactCommandTest.java:32]: wrong exception thrown, should be `DuplicateContactException`

**Review by goatygoatygoat — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/exceptions/CommandException.java (hunk 3 above)
Comment (nicholastanvis) [reply src/main/java/seedu/address/logic/commands/exceptions/CommandException.java]: explanation on telegram. resolved

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java (hunk 6 above)
Comment (KevinLohJunYong) [reply src/test/java/seedu/address/logic/commands/AddContactCommandTest.java]: Screenshot 2021-04-11 at 9 11 59 PM

File: src/test/java/seedu/address/logic/commands/AddContactCommandTest.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.contact.Contact;
+import seedu.address.testutil.ContactBuilder;
+
+public class AddContactCommandTest {
+    private ModelManager model = new ModelManager(getTypicalEntriesList(), new UserPrefs());
```
Comment (lrj689) [inline src/test/java/seedu/address/logic/commands/AddContactCommandTest.java]: This doesnt look like its being used so delete it

**Review by lrj689 — APPROVED** [review]

**Review by nicholastanvis — APPROVED** [review]

**Review by goatygoatygoat — APPROVED** [review]

**Review by tsh22 — APPROVED** [review]
